text/0018-genesis_block.md

```diff
@@ -0,0 +1,94 @@
+* Accounts from the testnet.
+
+# Token ownership
+We want Near investors and token holders to have their vesting schedule, lock up period and ability to stake to be implemented through the accounts and smart contracts on the blockchain.
+
+Each token holder will have tokens that go through the following lifecycle:
+* Unvested;
+* Vested, but locked;
+* Unlocked.
+
+Vesting is done over 4 years with 1 year cliff starting with:
+* For employees -- employment date or blockchain initiation date, whichever is latests;
+* For investors -- investment date.
+
+Non-standard vesting and lock up schedules can be later created by the Near Inc foundation.
+
+Lock up is applied to the token holder with 1 year starting with the mainnet launch. If token is vested before the lock up expires then it cannot be sold until the end of the lock up. If token is vested after the lock up then it can be sold immediately.
+
+In case of employees the NEAR Inc company can decide to terminate the employment and the person would lose the unvested tokens, however the lock period will still be applied to vested tokens.
+
+# Design details
+We create the following accounts and contracts:
+* For each token holder we create an account and a contract;
+* We create one account for NEAR Inc.
+
+## Token holder
+Token holder gets a standard account that has a special contract and balance corresponding to all vested and unvested tokens.
+The token holder person/entity gets one access key for this account, which allows to call the following functions:
+* `stake(amount)`;
+* `transfer(other_account_id, amount)`.
+
+`stake(amount)` simply issues a staking transaction from the account. It is allowed to use all even unvested tokens for staking.
+
+`transfer(other_account_id, amount)` can transfer vested locked tokens to a different account.
+
+Near Inc entity gets one access key for this account which allows to call these two functions:
+* `permanently_unstake()`;
+* `terminate()`.
+
+These functions will panic if account is not an employee (internal boolean flag). Otherwise
+
+`permanently_unstake()` will send unstaking transaction on behalf of the account and will switch the internal flag that will prohibit future calls to `stake(amount)`.
+
+`terminate()` will take the remaining unvested balance and send it to the NEAR Inc account.
+
+This access key will also have a limited allowance to avoid NEAR Inc calling it repeatedly to deplete the funds.
+
+Note, that no one in the system will have a full access key to the token holder account.
+
+Note, the contract will use the block timestamp to determine when the tokens are vested and locked.
+
+## NEAR Inc
+NEAR Inc will have a standard account with the balance of tokens that it has received from the foundation. The account will not have a contract deployed to it and there will only be one full access key given to it. It will have balance equal to the amount issued by the foundation minus all balances of the initial token holders.
```

Comment:
Hmm should this amount hold by "near foundation" instead? what is the difference

Reply:
Fixed.